boards/posix/simple_process/irq_ctrl.c

```diff
@@ -0,0 +1,271 @@
+/*
```

Comment:
Similar to my comment earlier, why can't this be in `arch/posix/`?

Reply:
Same about many of the files after this

Reply:
Please see my comment from the 5th of October above about the arch/soc/board layering in this arch. 
The idea is that the arch & SOC provide what almost everybody would want to reuse no matter what HW they are emulating. But if they want to, say, emulate an ARM's NVIC, they would include that NVIC model in their board.
( I do have another board with a different POSIX IRQ controller ).

Reply:
I read the comment and I'm not quite convinced. Why can't the different hardware models for interrupts live in subfolders in `arch/posix/`, just like we have for say, `arm/`, where we have `arm/core/` and then `arm/soc`?

Reply:
Ok. Let's separate the irq controller from the discussion one second.
The thing is: HW models did not exist in Zephyr or fit anywhere. 
So I placed the HW models for a board in boards/posix/<board_name>, and it's drivers in the drivers folder as for any other driver.
The idea in general is that, if for example, somebody models a given real peripheral, he can reuse the real driver from the drivers folder. But the HW model itself is not the driver, and I don't think it should be placed in the driver folder.
If you have some better idea where HW models should be placed please tell. But I guess it would need to be a completely new place, say native_hw_models/ , or something like that.
Please note that with time, I expect there will be many, very diverse hw models, based on different ways of modelling the HW, some meant for simulators or external tools, others like for the simple_process being just simple models.
Maybe we can just start how it is now, and if we see a need later we can move the hw models somewhere else.

The IRQ controller HW model is a bit special here because it is very linked with the SW IRQ handling (effectively its driver), which is normally part of the arch/soc. But in this case both the irq hw model and the irq handler are part of the board because of the layering explained in that other comment.